lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystemConfig.java

```diff
@@ -148,14 +163,92 @@ public GcsFileSystemConfig setJsonKeyFilePath(String jsonKeyFilePath)
         return this;
     }
 
+    public Optional<RetryStrategy> getRetryStrategy()
+    {
+        return this.retryStrategy;
+    }
+
+    @Config("gcs.retry.strategy")
+    @ConfigDescription("Determines the retry strategy: DEFAULT|UNIFORM. See https://cloud.google.com/storage/docs/retry-strategy.")
+    public GcsFileSystemConfig setRetryStrategy(RetryStrategy retryStrategy)
+    {
+        this.retryStrategy = Optional.ofNullable(retryStrategy);
+        return this;
+    }
+
+    public Optional<@MinDuration("0ms") Duration> getInitialRetryDelay()
+    {
+        return initialRetryDelay;
+    }
+
+    @Config("gcs.retry.initial-delay")
```

Comment:
We might want to name these to be consistent with the Thrift configs in `ThriftMetastoreConfig`

* `gcs.client.max-retries` -- note that we should do `setMaxAttempts(maxRetries + 1)` and have a minimum of zero for this
* `gcs.client.max-retry-time` -- this seems to be `setTotalTimeout()`
* `gcs.client.backoff-scale-factor`
* `gcs.client.min-backoff-delay`
* `gcs.client.max-backoff-delay`